packages/turf-transform-rotate/index.js

```diff
@@ -0,0 +1,114 @@
+var helpers = require('@turf/helpers');
+var point = helpers.point;
+var multiPoint = helpers.multiPoint;
+var lineString = helpers.lineString;
+var multiLineString = helpers.multiLineString;
+var polygon = helpers.polygon;
+var multiPolygon = helpers.multiPolygon;
+var feature = helpers.feature;
+var invariant = require('@turf/invariant');
+var getCoords = invariant.getCoords;
+var centroid = require('@turf/centroid');
+var rhumbBearing = require('@turf/rhumb-bearing');
+var rhumbDestination = require('@turf/rhumb-destination');
+var rhumbDistance = require('@turf/rhumb-distance');
+
+/**
+ * Rotates any geojson Feature or Geometry of a specified angle, around its `centroid` or a given `pivot` point;
+ * all rotations follow the right-hand rule: https://en.wikipedia.org/wiki/Right-hand_rule
+ *
+ * @name rotate
+ * @param {Geometry|Feature<any>} geojson object to be rotated
+ * @param {number} angle of rotation (along the vertical axis), from North in decimal degrees, negative clockwise
+ * @param {Geometry|Feature<Point>|Array<number>} [pivot=`centroid`] point around which the rotation will be performed
+ * @param {number} [xRotation=0] angle of rotation, respect to the horizontal (x,y) plain, about the x axis;
```

Comment:
Those are the (only?) way to support the `z` coordinate and rotations by the `x` and/or `y` axes; we'd need matrix operations for that though. No problem for me to add this feature, maybe after we have a first working version of the three `transform-` modules.
Having support of the third coordinate in some (e.g. `turf-transform-translate`) but not all the modules would be inconsistent/unexpected, wouldn't it?

In any way I still need to implement z-coordinate support (at least let it pass through the rotation "as is").